src/channel/error.rs

```diff
@@ -6,3 +6,67 @@
 /// The error recovers the value that has been sent.
 #[derive(PartialEq, Debug)]
 pub struct ChannelSendError<T>(pub T);
+
+/// The error which is returned when trying to receive from a channel
+/// without blocking fails.
+#[derive(PartialEq, Debug, Copy, Clone)]
+pub enum TryReceiveError {
+    /// The operation failed because it would block.
+    WouldBlock,
```

Comment:
`Block` is a bit confusing, because `Future`s never block :-) We could also call it along `NoElementAvailable` - but again too long.
`WouldBlock` is definitely the known equivalent from the synchronous world, so maybe we should just go for it.

Reply:
We could go for `Empty`.

Reply:
Sounds good. Or maybe `NotAvailable`.